.github/workflows/ci.yml

```diff
@@ -24,30 +24,69 @@ jobs:
         node-version: [12.x, 14.x]
 
     steps:
+      # Checkout
       - uses: actions/checkout@v2
       - name: Use Node.js ${{ matrix.node-version }}
         uses: actions/setup-node@v1
         with:
           node-version: ${{ matrix.node-version }}
 
+      # Caching: root project
+      - name: Get Yarn cache directory
+        id: yarn-cache-dir-path
+        run: echo "::set-output name=dir::$(yarn cache dir)"
+
+      # Note: Yarn root cache restore is slow (1:30) on Windows, so only do on Linux.
+      - name: Use Yarn cache
+        if: startsWith(runner.os, 'Linux')
+        uses: actions/cache@v2
+        id: yarn-cache
+        with:
+          path: ${{ steps.yarn-cache-dir-path.outputs.dir }}
+          key: yarn-${{ runner.os }}-${{ matrix.node-version }}-${{ hashFiles('**/yarn.lock') }}
+          restore-keys: |
+            yarn-${{ runner.os }}-${{ matrix.node-version }}-
+            yarn-${{ runner.os }}-
+
+      - name: Use node_modules cache
+        id: node-modules-cache
+        uses: actions/cache@v2
+        with:
+          path: node_modules
+          key: node-modules-${{ runner.os }}-${{ matrix.node-version }}-${{ hashFiles('./yarn.lock') }}
+          restore-keys: |
+              node-modules-${{ runner.os }}-${{ matrix.node-version }}-
+              node-modules-${{ runner.os }}-
+
       - name: Installation
-        run: yarn install --frozen-lockfile --non-interactive
+        run: yarn install --prefer-offline --frozen-lockfile --non-interactive
```

Comment:
In `urql` @kitten found a good way to improve this step, essentially by [checking whether there is a `cache-hit`](https://github.com/FormidableLabs/urql/blob/main/.github/workflows/ci.yml#L40-L42) we can skip this step entirely.

Reply:
Oh yeah! I saw that in the docs. I can use that here and in the other install too. I'll give it a whirl.